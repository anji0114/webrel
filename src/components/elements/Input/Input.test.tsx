import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { Input } from '.';

describe('Input interactions', () => {
  test('calls onChange handler when input', async () => {
    const handleChange = jest.fn();
    render(<Input onChange={handleChange} />);
    const inputElement = screen.getByRole('textbox');
    await userEvent.type(inputElement, 'Hello');
    expect(handleChange).toHaveBeenCalledTimes(5);
  });
});

test('Input snapshot', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
