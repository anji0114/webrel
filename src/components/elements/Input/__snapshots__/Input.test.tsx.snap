// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<input
  className="text-base border leading-none border-gray-400 rounded-lg bg-gray-200 outline-none focus:border-gray-500 focus:bg-gray-100 px-4 h-12"
/>
`;
